diff --git a/BLAS/SRC/Makefile b/BLAS/SRC/Makefile
index 486571fec..d413ce41e 100644
--- a/BLAS/SRC/Makefile
+++ b/BLAS/SRC/Makefile
@@ -146,8 +146,7 @@ ALLOBJ = $(SBLAS1) $(SBLAS2) $(SBLAS3) $(DBLAS1) $(DBLAS2) $(DBLAS3) \
 	$(ZBLAS2) $(ZBLAS3) $(ALLBLAS)
 
 $(BLASLIB): $(ALLOBJ)
-	$(AR) $(ARFLAGS) $@ $^
-	$(RANLIB) $@
+	$(FC) -z muldefs -shared -Wl,-soname,libblas.so -o $@ $(ALLOBJ)
 
 .PHONY: single double complex complex16
 single: $(SBLAS1) $(ALLBLAS) $(SBLAS2) $(SBLAS3)
